### PR TITLE
Add EAP XP 4.0

### PR DESCRIPTION
**Pull request by luck3y**
https://issues.redhat.com/browse/JBEAP-23231
Signed-off-by: kwills@redhat.com

### CODE REVIEW

File: official/eap/templates/eap-xp4-basic-s2i.json [hunk 1]
```diff
@@ -0,0 +1,531 @@
...
+					}
+				],
+				"publishNotReadyAddresses": true,
+				"selector": {
+					"deploymentConfig": "${APPLICATION_NAME}"
+				}
+			}
+		},
+		{
+			"apiVersion": "v1",
```
Comment (yselkowitz) [inline official/eap/templates/eap-xp4-basic-s2i.json:73]: Please fix in your upstream:
```suggestion
			"apiVersion": "route.openshift.io/v1",
```

File: official/eap/templates/eap-xp4-basic-s2i.json [hunk 2]
```diff
@@ -0,0 +1,531 @@
...
+			"kind": "ImageStream",
+			"metadata": {
+				"labels": {
+					"application": "${APPLICATION_NAME}"
+				},
+				"name": "${APPLICATION_NAME}-build-artifacts"
+			}
+		},
+		{
+			"apiVersion": "v1",
```
Comment (yselkowitz) [inline official/eap/templates/eap-xp4-basic-s2i.json:116]: Please fix in your upstream:
```suggestion
			"apiVersion": "build.openshift.io/v1",
```

Comment (yselkowitz) [inline official/eap/templates/eap-xp4-basic-s2i.json:116]: Please fix in your upstream:
```suggestion
			"apiVersion": "build.openshift.io/v1",
```

File: official/eap/templates/eap-xp4-basic-s2i.json [hunk 3]
```diff
@@ -0,0 +1,531 @@
...
+						"type": "ImageChange"
+					},
+					{
+						"type": "ConfigChange"
+					}
+				]
+			}
+		},
+		{
+			"apiVersion": "v1",
```
Comment (yselkowitz) [inline official/eap/templates/eap-xp4-basic-s2i.json:197]: Ditto.

Comment (yselkowitz) [inline official/eap/templates/eap-xp4-basic-s2i.json:197]: Ditto.

File: official/eap/templates/eap-xp4-basic-s2i.json [hunk 4]
```diff
@@ -0,0 +1,531 @@
...
```
Comment (yselkowitz) [inline official/eap/templates/eap-xp4-basic-s2i.json:257]: Please fix in your upstream:
```suggestion
			"apiVersion": "apps.openshift.io/v1",
```

Comment (yselkowitz) [inline official/eap/templates/eap-xp4-basic-s2i.json:257]: Please fix in your upstream:
```suggestion
			"apiVersion": "apps.openshift.io/v1",
```

**Review by yselkowitz — CHANGES_REQUESTED** [review]
Found some non-groupified API references that need to be fixed in your own repo and re-imported.

**Comment by dperaza4dustbit** [comment]
Hello @luck3y I see that EAP XP 3 is already added as a helm chart in our OpenShift Repository. Do you need it in the cluster as both sample and helm chart? What would be the use case for that? Depending on the answer can you consider adding EAP XP 4 as another helm chart in the certification repo: https://github.com/openshift-helm-charts/charts/tree/main/charts/redhat/redhat. These are the owners for EAP XP 3: https://github.com/openshift-helm-charts/charts/blob/main/charts/redhat/redhat/eap-xp3/OWNERS

**Comment by luck3y** [comment]
> Found some non-groupified API references that need to be fixed in your own repo and re-imported.

Thanks @yselkowitz, I thought I'd already checked those, will update.

**Comment by luck3y** [comment]
@dperaza4dustbit I'll need to get @jmesnil's feedback on what we should do here, thanks for raising the issue though.

**Comment by jmesnil** [comment]
@dperaza4dustbit Thanks for raising this issue. We are planning to put a Helm Chart for EAP XP 4 in OpenShift library soon (they are still in Tech Preview fwiw)
However we have not yet stop supporting OpenShift templates for EAP XP products, so we would like to have both available in OpenShift.

Please note this would be the last time we would have both: next versions of EAP will only provided (supported) Helm Charts.

**Comment by luck3y** [comment]
@gabemontero @yselkowitz This has been addressed in the source repo and updated, thanks for the feedback!

**Review by yselkowitz — APPROVED** [review]
/lgtm
/assign @dperaza4dustbit

**Pull request merged**